### PR TITLE
Using windows-sys instead winapi

### PR DESCRIPTION
**Pull request by akahan**
Using windows-sys instead winapi and use x25519_dalek instead of boringtun

### CODE REVIEW

**Comment by TroyNeubauer** [comment]
Looks like you've opened and closed several PR's here over the past few weeks without any discussion. I'm curious what you are doing here and how we can prevent extra work (open a PR just to close it without requesting a review).

**Comment by akahan** [comment]
Sorry for creating confusion and misunderstanding.  
I've been using this crate for a long time. I decided that someone would need my work. To avoid compiling unnecessary old winapi crates during compilation.
First I added unnecessary things to the PR, then deleted them.  Then I thought that there were too many extra commits, etc. 
Hopefully the PR is cleaner now.

**Comment by danielstuart14** [comment]
Tested here, was able to compile and run demo using latest windows 11 build. 
Winapi doesn't work anymore: https://github.com/paritytech/parity-tokio-ipc/issues/39

File: Cargo.toml [hunk 1]
```diff
@@ -19,14 +19,20 @@ targets = ["aarch64-pc-windows-msvc", "i686-pc-windows-msvc", "x86_64-pc-windows
...
+    "Win32_Networking_WinSock",
+    "Win32_NetworkManagement_IpHelper",
+    "Win32_NetworkManagement_Ndis"
+]}
 
 [dev-dependencies]
 env_logger = "0.9"
 base64 = "0.13"
-boringtun = "0.3"
+x25519-dalek = { version = "=2.0.0-rc.3", default-features = false, features = ["static_secrets", "getrandom"] }
```
Comment (TroyNeubauer) [inline Cargo.toml]: Any reason we cant use `2.0.1`?

File: Cargo.toml [hunk 2]
```diff
@@ -19,14 +19,20 @@ targets = ["aarch64-pc-windows-msvc", "i686-pc-windows-msvc", "x86_64-pc-windows
 widestring = "0.4"
 log = "0.4"
 rand = "0.8"
-bitflags = "1.3"
+bitflags = "2"
 
-libloading = "0.7"
+libloading = "0.8"
 ipnet = "2.3"
-winapi = { version = "0.3", features = ["ws2def", "netioapi", "nldef"] }
```
Comment (TroyNeubauer) [inline Cargo.toml:26]: nice

**Comment by TroyNeubauer** [comment]
Ah, thanks for flagging!

**Review by TroyNeubauer — COMMENTED** [review]
Overall - love the cleanup. One question about dep versioning

File: Cargo.toml (hunk 1 above)
Comment (akahan) [reply Cargo.toml]: I think no.  It’s just that where I use x25519-dalek there is a temporary limitation for this particular version, which I haven’t gotten around to fixing yet.
I'll replace it with 2.0.1 today.

Comment (akahan) [reply Cargo.toml]: There was either this or a similar error: https://github.com/nulldotblack/wireguard-nt/actions/runs/10126534546/job/28167328669?pr=15#step:3:117

File: Cargo.toml [hunk 3]
```diff
@@ -32,7 +32,7 @@ windows-sys = { version = "0.52", features = [
 ]}
 
 [dev-dependencies]
-env_logger = "0.9"
+env_logger = "0.11"
```
Comment (akahan) [inline Cargo.toml:35]: To get rid of winapi completely.

**Review by TroyNeubauer — APPROVED** [review]

**Pull request merged**